web/src/test/kotlin/no/nav/su/se/bakover/web/routes/søknad/SøknadRoutesKtTest.kt

```diff
@@ -170,4 +173,78 @@ internal class SøknadRoutesKtTest {
             }
         }
     }
+
+    @Test
+    fun `lager en søknad, så trekker søknaden`() {
+        withTestApplication({
+            testSusebakover()
+        }) {
+            val søknadCreateResponse = defaultRequest(
+                Post,
+                søknadPath
+            ) {
+                addHeader(ContentType, Json.toString())
+                setBody(soknadJson)
+            }.apply {
+                assertEquals(Created, response.status())
+            }.response
+
+            shouldNotThrow<Throwable> { objectMapper.readValue<SakJson>(søknadCreateResponse.content!!) }
+
+            val sak = sakRepo.hentSak(fnr)
+            sak shouldNotBe null
+            sak!!.søknader() shouldHaveAtLeastSize 1
+
+            defaultRequest(Post, "$søknadPath/${sak.søknader().first().id}/trekkSoknad") {
+                addHeader(ContentType, Json.toString())
+                setBody(
+                    """{
+                        "sakId": "${sak.id}",
+                        "søknadId": "${sak.søknader().first().id}",
+                        "søknadTrukket": true
+                    }
+                    """.trimIndent()
+                )
+            }.apply {
+                response.status() shouldBe OK
+            }.response
+        }
+    }
+
+    @Test
+    fun `en søknad som er trukket, skal ikke kunne bli trukket igjen`() {
+        withTestApplication({
+            testSusebakover()
+        }) {
+            val sak = sakRepo.opprettSak(fnr)
+            val søknad = søknadRepo.opprettSøknad(
+                sakId = sak.id,
+                søknad = Søknad(
+                    sakId = sak.id,
+                    søknadInnhold = build()
+                )
+            )
+            søknadRepo.trekkSøknad(
+                TrukketSøknadBody(
+                    sakId = sak.id,
+                    søknadId = søknad.id,
+                    søknadTrukket = true
+                )
+            )
+
+            defaultRequest(Post, "$søknadPath/${søknad.id}/trekkSoknad") {
+                addHeader(ContentType, Json.toString())
+                setBody(
+                    """{
+                        "sakId": "${sak.id}",
+                        "søknadId": "${søknad.id}",
+                        "søknadTrukket": true
+                    }
+                    """.trimIndent()
+                )
+            }.apply {
+                response.status() shouldBe InternalServerError
```

Comment:
Burde kanskje gi en bad request for slikt?

Reply:
👍